### PR TITLE
Add Reset password page with ability to reset password with password …

### PR DESCRIPTION
**Pull request by Timboh56**
…confirmation.

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -2,12 +2,13 @@
...
       <Toolbar></Toolbar>
       <!-- TODO: remove :key similar to https://github.com/whynotearth/shinta-mani-wild/pull/298 -->
-      <router-view :key="$route.name + ($route.params.id || '')"/> 
+      <router-view :key="$route.name + ($route.params.id || '')"/>
   </v-app>
 </template>
 
 <script>
 const Toolbar = () => import ('@/components/Toolbar.vue')
+const ResetPassword = () => import ('@/components/Auth/ResetPassword.vue')
```
Comment (mort3za) [inline src/App.vue]: We don't need this here. Please check the routes.js file and add a new route there.

File: src/components/Auth/ResetPassword.vue [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        </v-flex>
+      </v-row>
+      <v-row>
+        <v-flex xs12 fluid text-center>
+          <v-text-field
+            v-model="confirmPassword"
+            outlined
+            label="Confirm Password"
+            name="ConfirmPassword"
+            color="#B9BCC1"
```
Comment (mort3za) [inline src/components/Auth/ResetPassword.vue]: Instead of hardcoded color codes, we should use named colors. named colors can be found in themes.ts file.

File: src/components/Auth/ResetPassword.vue [hunk 3]
```diff
@@ -0,0 +1,118 @@
+<template>
+  <v-container align-center style="background: transparent;" fluid class="resetPasswordForm">
+    <v-flex xs12 md6 class='reset-password-flex mx-auto'>
+      <v-row>
+        <v-flex align-center xs12 fluid text-center>
```
Comment (mort3za) [inline src/components/Auth/ResetPassword.vue]: From the Vuetify docs:  
> v-col is a content holder that must be a direct child of v-row. This is the 2.x replacement for v-flex in 1.x.

File: src/connection/resources.js [hunk 4]
```diff
@@ -12,6 +12,13 @@ export class AuthenticationService {
   /**
```
Comment (mort3za) [inline src/connection/resources.js]: This file is auto-generated by a tool that get's data from Swagger. We don't need to add the API functions manually.

File: src/components/Auth/ResetPassword.vue [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+          </v-text-field>
+        </v-flex>
+      </v-row>
+      <v-row>
+        <v-flex xs12>
+          <v-btn
+            dark
+            large
+            depressed
+            class="menubtn mx-auto"
```
Comment (mort3za) [inline src/components/Auth/ResetPassword.vue]: please add .text-transform-none and .dark--text

Comment (mort3za) [reply src/components/Auth/ResetPassword.vue]: also please use BEM naming for custom class names. in this case: `reset-password--submit`

File: src/router/index.ts [hunk 6]
```diff
@@ -9,15 +9,26 @@ const Contact = () => import('../components/Contact.vue')
...
 import store from '@/store'
 
 const routes = [
   {
     path: '/',
+    props: true,
     component: Home
   },
+  {
+    path: '/reset-password',
```
Comment (mort3za) [inline src/router/index.ts]: `/auth/reset-password/:token` has better feelings I think.

File: src/router/index.ts [hunk 7]
```diff
@@ -9,15 +9,26 @@ const Contact = () => import('../components/Contact.vue')
...
+const ConfirmAndPay = () =>
+  import('../components/Reservation/ConfirmAndPay.vue')
 const KitchenSink = () => import('../views/KitchenSink.vue')
 
 import store from '@/store'
 
 const routes = [
   {
     path: '/',
+    props: true,
```
Comment (mort3za) [inline src/router/index.ts]: remove, please.

File: src/store/modules/auth.ts [hunk 8]
```diff
@@ -183,6 +183,9 @@ export default {
           }
         })
     },
+    resetPassword(context, resetToken) {
+      return AuthenticationService.resetPassword(resetToken)
```
Comment (mort3za) [inline src/store/modules/auth.ts]: use a console.log instead for now. we'll enable it when the API is ready.

File: src/store/modules/booking.ts [hunk 9]
```diff
@@ -111,6 +111,9 @@ export default {
     updateTransportation(state, payload) {
       state.bookingInfo.transportation = payload
     },
+    updateResetPassword(state, payload) {
```
Comment (mort3za) [inline src/store/modules/booking.ts]: This is booking store, so I think this file is unrelated to reset password. should be removed.

File: src/App.vue (hunk 1 above)
Comment (mort3za) [reply src/App.vue]: We have a `views` folder. every new route should be connected to a new component in the view folder, in that component we can import other components (reset-password component for example.).
So in views folder create a component: `ResetPasswordPage.vue` and import your component inside of it.

**Comment by mort3za** [comment]
I just noticed that you're editing a wrong file for adding your route, maybe because you didn't pulled the changes before starting to work on this branch.  
The router file is `src/router/index.ts` and routes.js file is removed.

**Comment by mort3za** [comment]
There is console errors on submit.  
Password rules don't work correctly, the error message should be updated when the passwords match.  
Follow the style of errors in the signup form, please.

Thank you

**Comment by mort3za** [comment]
Is there anything that I can help with? @Timboh56

**Comment by Timboh56** [comment]
@mort3za Not currently, I'm still waiting for the changes on the back end to support reset password. Thanks.
